src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -34,3 +41,11 @@ case class Concat(val left: RegularExpression, val right: RegularExpression)
  *  expression
  */
 case class Star(val expression: RegularExpression) extends RegularExpression
+
+/** the implicit definitions to add syntatic sugar so that Chars and Strings
+    can be used as regular expressions*/
+object ExpressionImplicits {
+  implicit def char2literal(c: Char): Literal = Literal(c)
+  implicit def string2literal (s: String): RegularExpression =
+    if (s.length == 0) EPSILON else Concat(Literal(s.charAt(0)), s.substring(1))
```

Comment:
Does this work? Don't you need to recursively call `string2literal` on `s.substring(1)`?